File: demos/ios/Demos/Demos/views/BasePointChart.m
//
//  BasePointUIView.m
//  F2Native
//
//  Created by 青原 on 2021/1/4.
//  Copyright © 2021 com.alipay.xgraph. All rights reserved.
//

#import "BasePointChart.h"

@implementation BasePointChart

- (void)chartRender {
    NSString *jsonPath = [[NSBundle mainBundle] pathForResource:@"Res/mockData_singlePointChart" ofType:@"json"];
    NSString *jsonData = [NSString stringWithContentsOfFile:jsonPath encoding:NSUTF8StringEncoding error:nil];
    self.chart.canvas(self.canvasView).padding(20, 10, 20, 0.f);
    self.chart.source([F2Utils toJsonArray:jsonData]);
    self.chart.point().position(@"x*y").size(@"z", @[@(3), @(10)]).fixedShape(@"circle");
    self.chart.animate(YES);
    self.chart.render();
}

@end

File: demos/ios/Demos-Swift/Demos-Swift/Views/BasePointChart.swift
//
//  BasePointChart.swift
//  F2Native
//
//  Created by Westin on 2024-06-09.
//

import F2

class BasePointChart: BaseLineChart {
    
    override func chartRender() {
        guard let jsonPath = Bundle.main.path(forResource: "Res/mockData_singlePointChart", ofType: "json"),
              let jsonDataString = try? String(contentsOfFile: jsonPath),
              let canvasView = self.canvasView,
              let chart = self.chart else {
            return
        }
        let jsonData = F2Utils.toJsonArray(jsonDataString)
        chart.clear()
        chart.canvas()(canvasView).padding()(20, 10, 20, 0)
        chart.source()(jsonData)
        chart.point()().position()("x*y").size()("z", [3, 10]).fixedShape()("circle")
        chart.animate()(true)
        chart.render()()
    }
}

